### PR TITLE
Mark activationRun as failed if there is an ContainerConnectionError

### PR DESCRIPTION
**Pull request by jiangpengcheng**
Just as the title said

## Related issue and scope
- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components
- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes
- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:
- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: common/scala/src/main/scala/whisk/core/containerpool/Container.scala [hunk 1]
```diff
@@ -142,12 +142,16 @@ trait Container {
...
       .andThen { // never fails
         case Success(r: RunResult) =>
-          transid.finished(
-            this,
-            start.copy(start = r.interval.start),
-            s"running result: ${r.toBriefString}",
-            endTime = r.interval.end,
-            logLevel = InfoLevel)
+          r.response match {
+            case Left(_) => transid.failed(this, start, s"run failed with ${r.toBriefString}", logLevel = ErrorLevel)
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: Can this be caused by user behavior or is this guaranteed to be an error of the system?

Comment (jiangpengcheng) [reply common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: I think so as the `ContainerConnectionError` here only occurred while connecting to action containers, or is there any thing users can do to terminate/prevent such connections?

Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: For blackbox actions for example, the run path can be broken.

File: common/scala/src/main/scala/whisk/core/containerpool/Container.scala [hunk 2]
```diff
@@ -142,12 +142,16 @@ trait Container {
     callContainer("/run", body, timeout, retry = false)
       .andThen { // never fails
         case Success(r: RunResult) =>
```
Comment (vvraskin) [inline common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: Did you consider adding a similar behaviour for the `/init` call as well? I see nearly the same code in https://github.com/apache/incubator-openwhisk/blob/master/common/scala/src/main/scala/whisk/core/containerpool/Container.scala#L102 

or maybe even extracting it into a separate method would help here...

File: common/scala/src/main/scala/whisk/core/containerpool/Container.scala (hunk 1 above)
Comment (jiangpengcheng) [reply common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: ok, but that should be marked as error too in my opinion , as the activation is failed anyway

Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/core/containerpool/Container.scala]: It is not an ERROR in the sense of the system is at fault though. Ideally, you'd want to monitor "ERROR" log messages in your production system and judge based on that if something is wrong. That loglevel should never be in the hand of the user of the system imo.

**Pull request closed without merging**